### DISCUSSION TITLE
req.user is undefined

### DISCUSSION THREAD

**Discussion by ricardonogues** [ericmakesapps/graphql-passport#88] [Q&A]
Hi,
I'm using this package to manage sessions with a PGSQL database. The login works fine and the user is correctly serialized to the DB. The problem seems to happen with getting the current user because req.user is undefined. Where is req.user being set? Is it something I have to do?

Thanks in advance.

**Comment by ericbf** [comment]
Have you set up the deserializer?

Something like:

```ts
passport.deserializeUser((id, cb) =>
	process.nextTick(async () => {
		const user = await User.findById(id).exec()

		cb(null, user : null)
	})
)
```

Have you set up your context for your operations? Are you using `context.getUser()`?
